lib/ocpp/v201/charge_point.cpp

```diff
@@ -220,6 +227,13 @@ void ChargePoint::on_firmware_update_status_notification(int32_t request_id,
     }
 
     if (this->firmware_status_before_installing == req.status) {
+        const auto transaction_active = this->any_transaction_active(std::nullopt);
+        if (transaction_active) {
+            this->firmware_status = FirmwareStatusEnum::InstallScheduled;
+            req.status = firmware_status;
+            ocpp::Call<FirmwareStatusNotificationRequest> call(req, this->message_queue->createMessageId());
+            this->send_async<FirmwareStatusNotificationRequest>(call);
+        }
```

Comment:
```suggestion
        // FIXME(Kai): This is a temporary workaround, because the EvseManager does not keep track of transactions.
        // If the firmware status is expected to become "Installing", but we still have a transaction running,
        // the update will wait for the transaction to finish, and so we send an "InstallScheduled" status.
        // This is necessary for OCTT TC_L_15_CS to pass.
        const auto transaction_active = this->any_transaction_active(std::nullopt);
        if (transaction_active) {
            this->firmware_status = FirmwareStatusEnum::InstallScheduled;
            req.status = firmware_status;
            ocpp::Call<FirmwareStatusNotificationRequest> call(req, this->message_queue->createMessageId());
            this->send_async<FirmwareStatusNotificationRequest>(call);
        }
```